scalafix-tests/unit/src/test/scala/scalafix/tests/core/SyntheticSuite.scala

```diff
@@ -0,0 +1,29 @@
+package scalafix.tests.core
+
+import System.{lineSeparator => nl}
+
+class SyntheticSuite extends BaseSemanticSuite("SyntheticTest") {
+
+  test("text") {
+    val synthetics = index.synthetics(input)
+    val obtained = synthetics.map(_.text).mkString(nl)
+
+    val expected =
+      """|*(scala.collection.immutable.List.canBuildFrom[scala.Int])
+         |*[scala.package.List]
+         |*[scala.Int, scala.Predef.Set[scala.Int]]
+         |*.scala.collection.immutable.List.apply[scala.Int]
+         |*(scala.collection.immutable.Set.canBuildFrom[scala.Int])
+         |*[scala.collection.immutable.List[scala.Int], scala.Int, scala.Predef.Set[scala.Int]]
```

Comment:
Originaly it was:
```scala
[List[Int], Int, Set[Int]](scala.collection.immutable.Set.canBuildFrom[Int])
```

It now produces two synthetics:
```scala
(scala.collection.immutable.Set.canBuildFrom[scala.Int])
[scala.collection.immutable.List[scala.Int], scala.Int, scala.Predef.Set[scala.Int]]
```

I will add a commit to merge the two and preserve the old synthetic.